src/test/python/mmlsparktest/vw/test_vw_cb.py

```diff
@@ -0,0 +1,190 @@
+# Prepare training and test data.
+import os
+import unittest
+import tempfile
+import pyspark
+
+from mmlsparktest.spark import *
+
+from mmlspark.vw import VowpalWabbitContextualBandit
+from mmlspark.vw import VowpalWabbitFeaturizer
+from mmlspark.vw import VectorZipper
+
+from pyspark.ml.tuning import ParamGridBuilder
+from pyspark.sql.types import *
+from pyspark.ml import Pipeline
+from pyspark.ml.wrapper import *
+from pyspark.ml.common import inherit_doc, _java2py, _py2java
+from pyspark.sql.utils import AnalysisException
+
+def has_column(df, col):
+    try:
+        df[col]
+        return True
+    except AnalysisException:
+        return False
+
+class VowpalWabbitSpec(unittest.TestCase):
+    def get_data(self):
+        # create sample data
+        schema = StructType([
```

Comment:
you can just pass the col names if you are okay with the standard schema inference

Reply:
Can you explain how I'd do this, I had a go and couldn't get any of the variations I tried to work?

Reply:
It seems as though by default it makes whole numbers Longs whereas it expects Integers. I can extend the internal conversions to convert Long to Int but then if the number is too large then it will be a conversion exception rather than a schema validation failure? Seems better to limit the schema

Reply:
no worries!